src/cmap/establish/handshake/mod.rs

```diff
@@ -35,7 +35,7 @@ struct ClientMetadata {
     application: Option<AppMetadata>,
     driver: DriverMetadata,
     os: OsMetadata,
-    platform: Option<String>,
```

Comment:
this only seemed to be an option to allow it to be initially set to `None` in `BASE_CLIENT_METADATA` below, and then conditionally set if `version_check::triple` returned a value. 
since we can unconditionally set that value now in one line, and the fact that we allowed this value to be `None` is kind of how this bug existed in the first place, it seems preferable to make it non-optional.